File: exp-7.py
import turtle

# Function to scale a point by given sx and sy
def scale_point(x, y, sx, sy):
    x_new = x * sx
    y_new = y * sy
    return x_new, y_new

# Function to scale a line by sx and sy
def scale_line(x1, y1, x2, y2, sx, sy):
    x1_new, y1_new = scale_point(x1, y1, sx, sy)
    x2_new, y2_new = scale_point(x2, y2, sx, sy)
    return x1_new, y1_new, x2_new, y2_new

# Function to scale a triangle by sx and sy
def scale_triangle(x1, y1, x2, y2, x3, y3, sx, sy):
    x1_new, y1_new = scale_point(x1, y1, sx, sy)
    x2_new, y2_new = scale_point(x2, y2, sx, sy)
    x3_new, y3_new = scale_point(x3, y3, sx, sy)
    return x1_new, y1_new, x2_new, y2_new, x3_new, y3_new

# Initialize screen and turtle
screen = turtle.Screen()
t = turtle.Turtle()

# Set up for drawing the original line and triangle
t.penup()

# Original line (x1, y1) to (x2, y2)
x1, y1, x2, y2 = -200, -100, 100, 100
# Original triangle (x1, y1), (x2, y2), (x3, y3)
x1_t, y1_t, x2_t, y2_t, x3_t, y3_t = -150, -150, 50, -150, 0, 50

# Scaling factors (sx, sy)
sx, sy = 1.5, 1.5  # Scale by 1.5x in both x and y directions

# Draw original line
t.goto(x1, y1)
t.pendown()
t.goto(x2, y2)
t.penup()

# Draw original triangle
t.goto(x1_t, y1_t)
t.pendown()
t.goto(x2_t, y2_t)
t.goto(x3_t, y3_t)
t.goto(x1_t, y1_t)
t.penup()

# Scale and draw the scaled line
scaled_line = scale_line(x1, y1, x2, y2, sx, sy)
t.goto(scaled_line[0], scaled_line[1])
t.pendown()
t.goto(scaled_line[2], scaled_line[3])
t.penup()

# Scale and draw the scaled triangle
scaled_triangle = scale_triangle(x1_t, y1_t, x2_t, y2_t, x3_t, y3_t, sx, sy)
t.goto(scaled_triangle[0], scaled_triangle[1])
t.pendown()
t.goto(scaled_triangle[2], scaled_triangle[3])
t.goto(scaled_triangle[4], scaled_triangle[5])
t.goto(scaled_triangle[0], scaled_triangle[1])
t.penup()

# Hide the turtle and display the result
t.hideturtle()
screen.mainloop()

File: exp-7.c
#include <graphics.h>
#include <conio.h>
#include <stdio.h>

// Function to scale a point by given sx and sy
void scale_point(int *x, int *y, float sx, float sy) {
    *x = *x * sx;
    *y = *y * sy;
}

// Function to scale a line by sx and sy
void scale_line(int *x1, int *y1, int *x2, int *y2, float sx, float sy) {
    scale_point(x1, y1, sx, sy);
    scale_point(x2, y2, sx, sy);
    line(*x1, *y1, *x2, *y2); // Draw scaled line
}

// Function to scale a triangle by sx and sy
void scale_triangle(int *x1, int *y1, int *x2, int *y2, int *x3, int *y3, float sx, float sy) {
    scale_point(x1, y1, sx, sy);
    scale_point(x2, y2, sx, sy);
    scale_point(x3, y3, sx, sy);
    // Draw scaled triangle
    line(*x1, *y1, *x2, *y2);
    line(*x2, *y2, *x3, *y3);
    line(*x3, *y3, *x1, *y1);
}

int main() {
    int gd = DETECT, gm;
    initgraph(&gd, &gm, ""); // Initialize graphics mode

    // Original line (x1, y1) to (x2, y2)
    int x1 = 100, y1 = 100, x2 = 300, y2 = 100;
    
    // Original triangle (x1, y1), (x2, y2), (x3, y3)
    int x1_t = 150, y1_t = 150, x2_t = 350, y2_t = 150, x3_t = 250, y3_t = 50;
    
    // Scaling factors (sx, sy)
    float sx = 1.5, sy = 1.5; // Scale by 1.5x in both x and y directions

    // Draw original line
    line(x1, y1, x2, y2);

    // Draw original triangle
    line(x1_t, y1_t, x2_t, y2_t);
    line(x2_t, y2_t, x3_t, y3_t);
    line(x3_t, y3_t, x1_t, y1_t);

    delay(2000); // Wait for 2 seconds before scaling

    // Scale and draw the scaled line
    scale_line(&x1, &y1, &x2, &y2, sx, sy);

    // Scale and draw the scaled triangle
    scale_triangle(&x1_t, &y1_t, &x2_t, &y2_t, &x3_t, &y3_t, sx, sy);

    getch(); // Wait for key press to exit
    closegraph(); // Close the graphics window
    return 0;
}
